Mathlib/Algebra/Lie/CartanExists.lean

```diff
@@ -0,0 +1,392 @@
+/-
+Copyright (c) 2024 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import Mathlib.Algebra.Lie.EngelSubalgebra
+import Mathlib.Algebra.Lie.CartanSubalgebra
+import Mathlib.Algebra.Lie.Rank
+import Mathlib.LinearAlgebra.Eigenspace.Zero
+
+/-!
+# Existence of Cartan subalgebras
+
+In this file we prove existence of Cartan subalgebras in finite-dimensional Lie algebras,
+following [barnes1967].
+
+## Main results
+
+* `exists_IsCartanSubalgebra_of_finrank_le_card`:
+  A Lie algebra `L` over a field `K` has a Cartan subalgebra,
+  provided that the dimension of `L` over `K` is less than or equal to the cardinality of `K`.
+* `exists_IsCartanSubalgebra`:
+  A finite-dimensional Lie algebra `L` over an infinite field `K` has a Cartan subalgebra.
+
+## References
+
+* [barnes1967]: "On Cartan subalgebras of Lie algebras" by D.W. Barnes.
+
+-/
+
+-- move this to `Mathlib.Algebra.MvPolynomial.Basic`
+lemma MvPolynomial.coe_aeval_eq_eval {R σ : Type*} [CommRing R] (x : σ → R) :
+    RingHomClass.toRingHom (MvPolynomial.aeval x) = MvPolynomial.eval x := rfl
+
+-- move this to `Mathlib.LinearAlgebra.FiniteDimensional`
+open FiniteDimensional in
+lemma _root_.Submodule.eq_top_of_finrank
```

Comment:
Isn't this [Submodule.eq_top_of_finrank_eq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/FiniteDimensional.html#Submodule.eq_top_of_finrank_eq)?

Reply:
Good catch!